src/components/page/endpoint.tsx

```diff
@@ -1,47 +1,96 @@
 
-	const showFullScreen = useIsMobileDevice();
-
 	const dispatch = useAppDispatch();
-	const onCloseModal = useCallback(() => dispatch(hideEndpointInfo()), [dispatch]);
 	const [
 		doShow,
-		{ hostname, port }
+		appEndpoint
 	] = useAppSelector((state: RootStateType) => [
 		getShowEndpointInfoModal(state),
 		getRunnerEndpoint(state)
 	]);
 
+	const [{ hostname, port }, setEndpoint] = useState<{ hostname: string; port: string; }>({ ...appEndpoint });
+	const [isEditing, setIsEditing] = useState<boolean>(false);
+	const showFullScreen = useIsMobileDevice();
+
+	const onCloseModal = useCallback(() => dispatch(hideEndpointInfo()), [dispatch]);
+
+	const onToggleEdit = useCallback((e: MouseEvent<HTMLButtonElement>) => {
+		if (isEditing) {
+			// Reset Values
+			setEndpoint({ ...appEndpoint });
+		}
+		setIsEditing(!isEditing);
+	}, [appEndpoint, isEditing, setIsEditing]);
+
+
+	const onChangeConfig = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+		setEndpoint({ hostname, port, [e.target.name]: e.target.value });
+	}, [hostname, port, setEndpoint]);
+
+	const onSubmit = useCallback((e: FormEvent<HTMLFormElement>) => {
+		e.preventDefault();
+		if (!isEditing) return;
+		window.location.href = `${window.location.protocol}//${window.location.host}?h=${encodeURIComponent(hostname)}&p=${encodeURIComponent(port)}`;
+	}, [isEditing, hostname, port]);
+
+	useEffect(() => {
+		if (!doShow && isEditing) {
+			setIsEditing(false);
+			setEndpoint({ ...appEndpoint });
+		}
+	}, [doShow, isEditing, appEndpoint, setEndpoint]);
+
 	return (
 		<Modal
 			onClose={ onCloseModal }
 			opened={ doShow }
 			fullScreen={ showFullScreen }
-			size="md"
+			size="lg"
 			title="OSCQuery Runner Endpoint"
 		>
-			<Stack gap="sm">
-				<TextInput
-					readOnly
-					label="Hostname"
-					description="The hostname or IP address of the device that runs the OSCQuery Runner"
-					value={ hostname }
-				/>
-				<TextInput
-					readOnly
-					label="Port"
-					description="The port of the OSCQuery Runner Websocket"
-					value={ port }
-				/>
-			</Stack>
+			<form onSubmit={ onSubmit } >
+				<Stack gap="md">
+					<TextInput
+						readOnly={ !isEditing }
+						onChange={ onChangeConfig }
+						name="hostname"
+						label="Hostname"
+						description="The hostname or IP address of the device that runs the OSCQuery Runner"
+						value={ hostname }
+						rightSection={ isEditing ? null : <ActionIcon variant="subtle" color="gray" onClick={ onToggleEdit }><FontAwesomeIcon icon={ faPencil } /></ActionIcon> }
+					/>
+					<TextInput
+						readOnly={ !isEditing }
+						onChange={ onChangeConfig }
+						name="port"
+						label="Port"
+						description="The port of the OSCQuery Runner Websocket"
+						value={ port }
+						rightSection={ isEditing ? null : <ActionIcon variant="subtle" color="gray" onClick={ onToggleEdit }><FontAwesomeIcon icon={ faPencil } /></ActionIcon> }
+					/>
```

Comment:
is there some markup that would make this numeric, for instance on mobile would launch the numeric keyboard instead of alpha?

Reply:
yeah `inputmode` should do that. Just adjusted for that